source/slang/emit.cpp

```diff
@@ -5997,6 +6009,19 @@ struct EmitVisitor
         
         emit(") ");
 
+        /*
+        If the output type is a buffer, and we can determine it is only readonly we can prefix before
+        buffer with 'readonly'
+
+        The actual structuredBufferType could be
+
+        HLSLStructuredBufferType                        - This is unambiguously read only
+        HLSLRWStructuredBufferType                      - Read write
+        HLSLRasterizerOrderedStructuredBufferType       - Allows read/write access
+        HLSLAppendStructuredBufferType                  - Write
+        HLSLConsumeStructuredBufferType                 - TODO (JS): Its possible that this can be readonly, but we currently don't support on GLSL
```

Comment:
Good comment about the append/consume case. Supporting those on VK would be a nice-to-have feature, which would require desugaring them into a combination of an ordinary structured buffer plus an atomic counter (since the atomic counter isn't implicitly part of the buffer type like in D3D). We are set up reasonably well to handle the layout part of that correctly (declaring that a parameter consumes two different kinds of resources is not a challenge for our approach), but the actual code generation would be annoying (do we try to handle in via stdlib code, or via an IR transformation? each has its benefits/drawbacks).